### PR TITLE
Dialog page initial commit

### PR DESCRIPTION
**Pull request by Nicelx**

### CODE REVIEW

File: src/pages/dialog/Dialog.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+	
+	sendMessageHandler = (objToPush) => {
+		console.log(objToPush);
+		const copy = [...this.state.messagesData];
+		copy.push(objToPush);
+		this.setState({
+			messagesData: copy
+		}
```
Comment (AlexBychkov) [inline src/pages/dialog/Dialog.js]: it's better to right this way ' }) '

File: src/pages/dialog/components/MessageItemNotYours.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import classes from './MessageItem.module.css'
+
+const MessageItemNotYours = props => {
+	return (
+		<div className = {classes.NotYourMessage}>
+			<div className = {classes.NotYourMessageLeftBlock}>
+				<span style = {{fontWeight: 100, fontSize: '0.7em'}}>{props.name}  {props.time}</span>
```
Comment (AlexBychkov) [inline src/pages/dialog/components/MessageItemNotYours.js]: just a reminder to move style to css file

File: src/pages/dialog/components/MessageItem.module.css [hunk 3]
```diff
@@ -0,0 +1,53 @@
+img {
+	border-radius: 50%;
+	width: 50px;
+	height: 50px;
+}
```
Comment (foxprogs) [inline src/pages/dialog/components/MessageItem.module.css:6]: add object-fit: cover; to disable change image proportions

File: src/pages/dialog/Dialog.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		)	
+	}
```
Comment (foxprogs) [inline src/pages/dialog/Dialog.js:66]: May be after send message clear textarea and scroll down to new message

File: src/pages/dialog/components/MessageItem.module.css [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+.YourMessage {
```
Comment (foxprogs) [inline src/pages/dialog/components/MessageItem.module.css:9]: For fix design change this to 
max-width: 40vw;
display: flex;
margin-left: 5vw;

File: src/pages/dialog/components/MessageItem.module.css [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	max-width: 30vw;
+	margin: 5px 0px 5px 5px;
+	padding: 5px;
+	background-color: white;
+	border: 2px solid #f1f3f4;
+	border-radius: 10px 10px 10px 0px;
+}
+
+
+.NotYourMessage {
```
Comment (foxprogs) [inline src/pages/dialog/components/MessageItem.module.css:36]: For fix design change this to 
max-width: 40vw;
display: flex;
margin-right: 5vw;
align-self: end;

File: src/pages/dialog/Dialog.module.css [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	margin-bottom: 10px;
+	height: 95vh;
+}
+
+/*  WE HAVE BUG HERE BELOW      !!!!!!!!!!!!!!!!!!!!! */
+/*  WE HAVE BUG HERE BELOW      !!!!!!!!!!!!!!!!!!!!! */
+/*  WE HAVE BUG HERE BELOW      !!!!!!!!!!!!!!!!!!!!! */
+/*  WE HAVE BUG HERE BELOW      !!!!!!!!!!!!!!!!!!!!! */
+/* overflow property doesnt works perfectly(((( */
+.DialogField {
```
Comment (foxprogs) [inline src/pages/dialog/Dialog.module.css:13]: For fix design change this to
display: flex;
flex-direction: column;
background-color: white;
padding: 20px;
overflow-y: scroll;
max-height: infinite;

File: src/pages/dialog/components/MessageItemYours.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+const MessageItemYours = props => {
+	return (
+		<div className = {classes.YourMessage}>
+			<div>
+				<img src={props.avatar} alt = 'avatar'/>
+			</div>
+			
+			<div className = {classes.YourMessageRightBlock}>
+				<span style = {{fontWeight: 100, fontSize: '0.7em'}}>{props.name}  {props.time}</span>
```
Comment (foxprogs) [inline src/pages/dialog/components/MessageItemYours.js]: just a reminder to move style to css file  =)

File: src/pages/dialog/components/MessageItem.module.css [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+.YourMessageRightBlock {
+	display: flex;
+	flex-direction: column;
+}
+.YourMessageRightBlock span {
+	text-align: start;
+	margin-left: 20px;
+}
+.YourMessageRightBlock p {
+	max-width: 30vw;
```
Comment (foxprogs) [inline src/pages/dialog/components/MessageItem.module.css:26]: for display multiline text add this style
white-space: pre-wrap;

File: src/pages/dialog/Dialog.module.css [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+	margin-bottom: 10px;
+	height: 95vh;
+}
+
+
+/* overflow property doesnt works perfectly... */
+.DialogField {
+	display: flex;
+	flex-direction: column;
+	justify-content: center;
```
Comment (foxprogs) [inline src/pages/dialog/Dialog.module.css]: in Firefox if remove this line - scroll work perfect, otherwise some messages in top of list is hidden

**Review by foxprogs — APPROVED** [review]
need one little change in css

File: src/pages/dialog/Dialog.module.css (hunk 10 above)
Comment (Nicelx) [reply src/pages/dialog/Dialog.module.css]: ty so much!!! google chrome works perfectly either -1 problem)))

File: src/pages/dialog/components/InputTextArea.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+		if (this.state.inputValue !== '') this.props.sendMessage(this.createMessageObjectClickHandler(),e)
+	}
+
+	// Very small bug here below 
+	// We don't have clear input after this handler. 
+	// And because of that    if (this.state.inputValue !== '') 
+	// pass handler when input have some sort of 'Enter key' input  'clear but not clear'
+	// i'll fix this soon
+	sendMessageOnKeyDownHandler = (e) => {
+		if (this.state.inputValue !== '')
```
Comment (foxprogs) [inline src/pages/dialog/components/InputTextArea.js:35]: May be remove this  "if" because in 26 line you check this again

**Review by AlexBychkov — APPROVED** [review]

**Pull request merged**